src/Microsoft.Kusto.ServiceLayer/DataSource/Monitor/MonitorDataSource.cs

```diff
@@ -108,8 +108,31 @@ public override IEnumerable<DataSourceObjectMetadata> GetChildObjects(DataSource
 
             if (parentMetadata.MetadataType == DataSourceMetadataType.Cluster && includeSizeDetails)
             {
-                var child = _nodes[parentMetadata.Urn].FirstOrDefault();
-                return child == null ? Enumerable.Empty<DataSourceObjectMetadata>() : _nodes[child.Urn];
+                var tableGroups = _nodes[parentMetadata.Urn];
+
+                if (tableGroups == null)
+                {
+                    return Enumerable.Empty<DataSourceObjectMetadata>();
+                }
+
+                var results = new SortedList<string, DataSourceObjectMetadata>(StringComparer.OrdinalIgnoreCase);
+                foreach (var tableGroup in tableGroups)
+                {
+                    var tables = _nodes[tableGroup.Urn];
+                    foreach (var table in tables)
+                    {
+                        if (results.ContainsKey(table.PrettyName))
```

Comment:
Probably better to do this with a Dictionary. ContainsKey is a O(Log n) operation for SortedList, and approaches  O(1) for  Dictionary. Dictionary will be much faster for a large number of tables.